pkg/apitypes/managed_tx.go

```diff
@@ -125,3 +153,37 @@ type TransactionUpdateReply struct {
 	ProtocolID      string       `json:"protocolId"`
 	TransactionHash string       `json:"transactionHash,omitempty"`
 }
+
+// Transaction sub-status entries can be added for a given transaction so a caller
+// can see discrete steps in a transaction moving to confirmation on the blockchain.
+// There only exists a single entry in the list for each unique sub-status type. If
+// a transaction goes through the same sub-status more than once (for example if it
+// has gas recalculated for it more than once) then the "Count" and "LastOccurence" fields
+// should be updated accordingly. This design allows a caller to see when the most recent
+// sub-status changes took place and if necessary, to order them by time, but prevents
+// the potential for the sub-status list to grow indefinitely.
+func (mtx *ManagedTX) AddSubStatus(ctx context.Context, subStatus TxSubStatus) {
+	// See if this status exists in the list already
+	for _, entry := range mtx.SubStatusHistory {
+		if entry.Status == subStatus {
+			entry.Count++
+			entry.LastOccurrence = fftypes.Now()
+			return
+		}
+	}
+
+	// Prevent crazy run-away situations by limiting the number of unique sub-status
+	// types we will keep
+	if len(mtx.SubStatusHistory) >= 50 {
```

Comment:
Assuming we retain a limit on the buffer, I think we need to:
1. Make the limit configurable
2. Discard the oldest record, not the newest